### PR TITLE
improve readability of blockchain's _update method

### PR DESCRIPTION
**Pull request by pcppcp**
splits `_update` method into several easily manageable parts

### CODE REVIEW

File: raiden_libs/blockchain.py [hunk 1]
```diff
@@ -272,3 +225,62 @@ def _update(self):
 
         if not self.wait_sync_event.is_set() and new_unconfirmed_head_number == current_block:
             self.wait_sync_event.set()
+
+    def filter_events(self, filter_params: dict, name_to_callback: dict) -> int:
+        """
```
Comment (palango) [inline raiden_libs/blockchain.py]: no newline

File: raiden_libs/blockchain.py [hunk 2]
```diff
@@ -272,3 +225,62 @@ def _update(self):
...
+        Params:
+            filter_params: arguments for the filter call
+            name_to_callback: dict that maps event name to callbacks executed
+                if the event is emmited
+        """
```
Comment (palango) [inline raiden_libs/blockchain.py]: no newline either

File: raiden_libs/blockchain.py [hunk 3]
```diff
@@ -272,3 +225,62 @@ def _update(self):
...
+                **filter_params
+            )
+            for event in events:
+                log.debug('Received confirmed %s event', event_name)
+                callback(event)
+
+    def reset_unconfirmed_on_reorg(self, current_block):
+        """Test if chain reorganization happened (head number used in previous pass is greater than
+        current_block parameter) and in that case reset unconfirmed event list.
+        """
```
Comment (palango) [inline raiden_libs/blockchain.py]: no newline

File: raiden_libs/blockchain.py [hunk 4]
```diff
@@ -272,3 +225,61 @@ def _update(self):
...
+            )
+            for event in events:
+                log.debug('Received confirmed %s event', event_name)
+                callback(event)
+
+    def reset_unconfirmed_on_reorg(self, current_block):
+        """Test if chain reorganization happened (head number used in previous pass is greater than
+        current_block parameter) and in that case reset unconfirmed event list."""
+        if self.wait_sync_event.is_set():  # but not on first sync
+            if current_block < self.unconfirmed_head_number:
```
Comment (hackaugusto) [inline raiden_libs/blockchain.py]: I believe the block number is not sufficient to detect reorgs, any plans to adjust and use the blockhash?

Comment (palango) [reply raiden_libs/blockchain.py]: Opened #25 to track that.

**Pull request merged**